site/content/docs/howto/configure-supervisor-with-dex.md

```diff
@@ -0,0 +1,132 @@
+
+In this example, we'll show how to use Dex to identify users through their GitHub account.
+
+First, we need to go to your Github account settings and [create an OAuth app](https://github.com/settings/applications/new) by populating the following rows -
+
+- Application name - `Dex application`
+- Homepage URL - `https://<dex-dns-record>`
+- Authorization callback URL - `https://<dex-dns-record>/callback` // this is where Github will redirect you to once your app has authorized
+
+Once completed, copy your `Client ID` and `Client secret` (generate one if there's none) as those two will be needed to configure a Github connector in Dex.
+
+To setup one, edit the configuration used by Dex by adding the following -
+
+```bash
+...
+connectors:
+- type: github
+  id: github
+  name: GitHub
+  config:
+    clientID: $GITHUB_CLIENT_ID
+    clientSecret: $GITHUB_CLIENT_SECRET
+    redirectURI: https://<dex-dns-record>/callback
+...
+```
+
+## Register an application in Dex
+
+Follow the instructions for [registering an application in Dex](https://dexidp.io/docs/using-dex/#configuring-your-app) and create a static client application, in our case the client happens be the Supervisor. Note that the "openid" scope is always included, but you can always request additional scopes that you can then pass to your Kubernetes cluster, such as "groups" for example.
+
+To create a static client application, edit the configuration used by Dex (can be a file or a ConfigMap) by adding the following -
+
+```bash
+...
+staticClients:
+- id: pinniped-supervisor
+  secret: pinniped-supervisor-secret
+  name: 'Pinniped Supervisor client'
+  redirectURIs:
+  - 'http://<pinniped-supervisor-dns-record>/callback'
+...
+```
+
+## Configure the Supervisor
+
+Create an [OIDCIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.20/README.adoc#oidcidentityprovider) resource in the same namespace as the Supervisor.
+
+For example, the following OIDCIdentityProvider and the corresponding Secret use Dex's `email` claim as the Kubernetes username:
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: OIDCIdentityProvider
+metadata:
+  namespace: pinniped-supervisor
+  name: dex
+spec:
+  # Specify the upstream issuer URL (no trailing slash).
+  issuer: https://<dex-dns-record>
+
+  # Request any scopes other than "openid" for claims besides
+  # the default claims in your token. The "openid" scope is always
+  # included.
+  authorizationConfig:
+    additionalScopes: [groups, email]
+
+  # Specify how Dex claims are mapped to Kubernetes identities.
+  claims:
+    # Specify the name of the claim in your Dex ID token that will be mapped
+    # to the "username" claim in downstream tokens minted by the Supervisor.
+    username: email
+
+    # Specify the name of the claim in your Dex ID token that represents the groups
+    # that the user belongs to. This matches what you specified above
+    # with the Groups claim filter.
+    groups: groups
```

Comment:
Do you actually get groups out of github? I'm reading through [this Dex doc](https://dexidp.io/docs/connectors/github/) and it seems like a lot of configuration that isn't in this clean, simple example. Perhaps we should make note of that so people aren't surprised to get no groups.

Reply:
@margocrawf It should be possible to receive group claims in the form of "org:team" if the user configured a Github organization to search against. But you're right though that in this case, this might not be applicable, so it'd be better to note it. Thanks for pointing that out! 👍